src/main/java/org/folio/rest/persist/HelperUtils.java

```diff
@@ -155,10 +155,12 @@ public static List<Error> buildNullValidationError(String value, String key) {
    * @param  method  action that will be executed sequentially based on the number of list items
    * @return         the last composed element(Feature result)
    */
-  public static <T, R> Future<R> chainCall(List<T> list, Function<T, Future<R>> method){
-    return list.stream().reduce(Future.succeededFuture(),
-      (acc, item) -> acc.compose(v -> method.apply(item)),
-      (a,b) -> Future.succeededFuture());
+  public static <T, R> Future<R> chainCall(List<T> list, Function<T, Future<R>> method) {
```

Comment:
I think the same thing was being done here. But if you think it's easier to perceive the method I don't mind.

Reply:
I am concerned about [this issue](https://stackoverflow.com/questions/29210176/can-a-collectors-combiner-function-ever-be-used-on-sequential-streams).